src/cudamatrix/cu-device.h

```diff
@@ -55,11 +55,14 @@ class CuDevice {
   inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
 
   inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
-    if (random_stride_mode_) {
+    if (debug_stride_mode_) {
       // The pitch bucket size is hardware dependent.
       // It is 512 on K40c with CUDA 7.5
-      return allocator_.MallocPitch(row_bytes + 512 * (rand() & 1), num_rows,
-                                    pitch);
+      // "% 8" ensures that any 8 adjacent allocations have different pitches
+      // if their original pitches are same in the normal mode.
+      return allocator_.MallocPitch(
+          row_bytes + 512 * ((num_debug_stride_allocations_++) % 8), num_rows,
```

Comment:
The way you are doing it here will use a potentially very large amount of memory, because it's using on average 512 * 4 extra bytes per row, and what if it's a matrix with a lot of rows?  Does it have to be % 8?